src/community/sldService/src/test/java/org/geoserver/sldservice/rest/RasterizerTest.java

```diff
@@ -0,0 +1,166 @@
+/* (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * Copyright (C) 2007-2008-2009 GeoSolutions S.A.S.
+ *  http://www.geo-solutions.it
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.sldservice.rest;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+
+import org.geoserver.catalog.LayerInfo;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.rest.RestBaseController;
+import org.geotools.styling.ColorMap;
+import org.geotools.styling.ColorMapEntry;
+import org.geotools.styling.RasterSymbolizer;
+import org.geotools.styling.Rule;
+import org.junit.Test;
+import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.w3c.dom.Document;
+
+public class RasterizerTest extends SLDServiceBaseTest {
+
+	ClassPathXmlApplicationContext appContext;
+	
+	@Override
+	protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+		appContext = new ClassPathXmlApplicationContext(
+				"classpath:**/sldservice-applicationContext.xml");
+		appContext.refresh();
+		//new GeoServerExtensions().setApplicationContext(appContext);
+		for(BeanFactoryPostProcessor postProcessor : appContext.getBeanFactoryPostProcessors()) {
+			applicationContext.addBeanFactoryPostProcessor(postProcessor);
+		}
+		applicationContext.refresh();
+
+		getTestData().addWorkspace(getTestData().WCS_PREFIX, getTestData().WCS_URI, getCatalog());
+		getTestData().addDefaultRasterLayer(getTestData().WORLD, getCatalog());
+	}
+
+	/* (non-Javadoc)
+	 * @see org.geoserver.test.GeoServerSystemTestSupport#onTearDown(org.geoserver.data.test.SystemTestData)
+	 */
+	@Override
+	protected void onTearDown(SystemTestData testData) throws Exception {
+		super.onTearDown(testData);
+		appContext.close();
+	}
+
+	@Test
+	public void testRasterizeWithNoParams() throws Exception {
+		LayerInfo l = getCatalog().getLayerByName("wcs:World");
+        assertEquals( "raster", l.getDefaultStyle().getName() );
+        final String restPath = RestBaseController.ROOT_PATH + "/sldservice/wcs:World/"+getServiceUrl()+".xml";
+        MockHttpServletResponse response = getAsServletResponse(restPath);
+        // Randomly cannot find REST path
+        if(response.getStatus() == 200) {
+	        Document dom = getAsDOM(restPath, 200);
+	        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+	        // print(dom);
+	        print(dom, baos);
+			assertTrue(baos.toString().indexOf("<featureTypeStyles>")>0);
+			// checkColorMap(baos.toString(), 5);
+        }
+	}
+
+//	public void testRasterizeWithFeatureType() throws IOException {
```

Comment:
Did the code coverage go down here? It's community module, stricly speaking it could have no test and be jolly good anyways, but seeing commented out tests is a potential sign of regressions (which again might be fine, a community module is unsupported in any and every way). Still... could you comment on this?

Reply:
Currently since this is a community module very old, and those methods other than "attributes" have not been used since a long time, I would prefer to open a specific PR for re-adding again tests to the module, since they must be rewritten from scratch basically.